src/snowflake/snowpark/dataframe.py

```diff
@@ -165,10 +170,26 @@ def _get_unaliased(col_name: str) -> List[str]:
     return unaliased
 
 
-def _alias_if_needed(df: "DataFrame", c: str, prefix: str, common_col_names: List[str]):
+def _alias_if_needed(
+    df: "DataFrame",
+    c: str,
+    prefix: Optional[str],
+    suffix: Optional[str],
+    common_col_names: List[str],
+):
     col = df.col(c)
     unquoted = c.strip('"')
     if c in common_col_names:
+        if suffix:
+            column_case_insensitive = is_snowflake_quoted_id_case_insensitive(c)
+            suffix_unqouted_case_insensitive = (
+                is_snowflake_unquoted_suffix_case_insensitive(suffix)
+            )
+            return col.alias(
+                f'"{unquoted}{suffix.upper()}"'
+                if column_case_insensitive and suffix_unqouted_case_insensitive
+                else f'''"{unquoted}{escape_quotes(suffix.strip('"'))}"'''
```

Comment:
When `suffix_unqouted_case_insensitive` is `False`, it could also mean that the suffix is not a valid suffix. Should we handle it?

Reply:
An "invalid" ID will be quoted then it becomes valid. It's hard to validate the length here so we'll let the Snowflake DB to raise an error if it's too long.